src/appleseed/renderer/modeling/input/maxsource.h

```diff
@@ -0,0 +1,154 @@
+#ifndef APPLESEED_RENDERER_MODELING_INPUT_MAXSOURCE_H
+#define APPLESEED_RENDERER_MODELING_INPUT_MAXSOURCE_H
+
+// appleseed.renderer headers.
+#include "renderer/global/globaltypes.h"
+#include "renderer/modeling/color/colorspace.h"
+#include "renderer/modeling/input/source.h"
+
+// Standard headers.
+#include <cstddef>
+#include <vector>
+
+// appleseed.main headers.
+#include "main/dllsymbol.h"
+
+namespace renderer
+{
+    class APPLESEED_DLLSYMBOL MaxSource
```

Comment:
I actually wonder why you need this class at all, since all it does is add a `sample_max_texture()` method.

What link errors did you get when you tried to inherit from `renderer::Source` in appleseed-max?

Reply:
Were they due to `renderer::TextureCache` maybe?

Reply:
Hi Franz,
It was `g_std_lighting_conditions` which is declared as `extern` in modeling/color/colorspace.h. It carries a lot of variables and constants in order to be initialized.
I tried different things to have that variable visible to max plugin but could fix it. I've tried to include corresponding colospace.cpp file, tried to copy/paste everything to have that variable defined in plugin scope but that didn't help either.
The best I get is successful compilation but link was still complaining.